### DISCUSSION TITLE
[FEATURE REQUEST] Longer domain handling in salt-cloud digitalocean provider

### DISCUSSION THREAD

**Discussion by waynegemmell** [saltstack/salt#67492] [Feature Requests]
The digitalocean provider only handles domains that have two parts. Our domains have three parts, xyz.co.za. The domain name that is setup is then co.za instead of xyz.co.za

Currently the code looks like this. 

```
default_dns_hostname = ".".join(dns_domain_name[:-2])
default_dns_domain = ".".join(dns_domain_name[-2:])
```
It should use length - 1 instead of "-2:" to ensure that the whole domain is captured.

**Comment by morsik** [comment]
It's 2024 and I'm still hitting this basic bug... :/

**Comment by morsik** [comment]
@dwoz why this was moved to Discussion? There's nothing to discuss about, it's a bug that should be an issue...

**Reply by dwoz** [reply]
No one is actively working the issue/feature for any current release. If/when someone is going to work on getting it fixed in a release we will re-open an issue.
